troi/tools/spotify_lookup.py

```diff
@@ -170,3 +171,56 @@ def submit_to_spotify(spotify, playlist, spotify_user_id: str, is_public: bool =
     playlist.add_metadata({"external_urls": {"spotify": playlist_url}})
 
     return playlist_url, playlist_id
+
+
+def get_tracks_from_playlist(spotify_token, playlist_id):
+    """ Get tracks from the Spotify playlist.
+    """
+    sp = spotipy.Spotify(auth=spotify_token, requests_timeout=10, retries=10)
+    playlist_info = sp.playlist(playlist_id)
+    playlists = sp.playlist_items(playlist_id, limit=100)
+    name = playlist_info["name"]
+    description = playlist_info["description"]
+    
+    return playlists, name, description
+
+def _convert_tracks_to_json(tracks_from_playlist):
+    tracks= []
+    for track in tracks_from_playlist["items"]:
+        artists = track['track'].get('artists', [])
+        artist_names = []
+        for a in artists:
+            name = a.get('name')
+            if name is not None:
+                artist_names.append(name)
+        artist_name = ', '.join(artist_names)
+        tracks.append({
+            "track_name": track['track']['name'],
+            "artist_name": artist_name,
+        })
+    return tracks
+
+def music_service_tracks_to_mbid(token, playlist_id):
+    """ Convert Spotify playlist tracks to a list of MBID tracks.
+    """
+    tracks_from_playlist, name, desc = get_tracks_from_playlist(token, playlist_id)
+    tracks = _convert_tracks_to_json(tracks_from_playlist)
+
+     # select track_name and artist_name for each track
+    mbid_mapped_tracks = [mbid_mapping_spotify(track["track_name"], track["artist_name"]) for track in tracks]
```

Comment:
I have added a bulk lookup endpoint to reduce the number of API calls. It supports looking upto 50 tracks in 1 API call. The endpoint is documented at https://listenbrainz.readthedocs.io/en/latest/users/api/metadata.html#post--1-metadata-lookup-.

You should chunk the tracks list into batches of 50 and perform an API call on each batch then join the result of all batches.